src/common/core/metacore.js

```diff
@@ -686,8 +794,93 @@ define([
 
             return false;
         };
+
+        this.getAttributeDefinitionOwner = function (node, name) {
+            while (node) {
+                if (self.getOwnValidAttributeNames(node).indexOf(name) !== -1) {
+                    return node;
+                }
+                node = self.getBase(node);
+            }
+
+            return null;
+        };
+
+        this.getPointerDefinitionInfo = function (node, name, target) {
+            return getRelationDefinitionInfo(node, name, target);
+        };
+
+        this.getSetDefinitionInfo = function (node, name, target) {
+            return getRelationDefinitionInfo(node, name, target);
+        };
+
+        this.getAspectDefinitionInfo = function (node, name, target) {
+            var result = {},
+                validTargets,
+                definedTarget,
+                targetPath;
+
+            while (node) {
+                if (self.getOwnValidAspectNames(node).indexOf(name) !== -1) {
+                    result.sourcePath = self.getPath(node);
+                    result.sourceNode = node;
+                    validTargets = self.getOwnValidAspectTargetPaths(node, name);
+                    definedTarget = target;
+                    while (definedTarget) {
+                        targetPath = self.getPath(definedTarget);
+                        if (validTargets.indexOf(targetPath) !== -1) {
+                            result.targetPath = targetPath;
+                            result.targetNode = definedTarget;
+                            return result;
+                        }
+                    }
+                }
+            }
+            return null;
+        };
+
+        this.getAspectDefinitionOwner = function (node, name) {
+
+            while (node) {
+                if (self.getOwnValidAspectNames(node).indexOf(name) !== -1) {
+                    break;
+                }
+
+                node = self.getBase(node);
+            }
+
+            return node;
+        };
+
+        this.getChildDefinitionInfo = function (node, child) {
+            var result = {},
+                validPaths,
+                definedChild,
+                childPath;
+
+            while (node) {
+                if (self.isValidChildOf(child, node)) {
+                    validPaths = self.getOwnValidChildrenPaths(node);
+                    result.souceNode = node;
```

Comment:
Maybe combine all the four cases and pass in the `isValid*` and `getOwnValid*` functions. (We now have `isValidAspectMemberOf `)